### PR TITLE
Migrate new UI to Chakra v3

### LINKED ISSUE #43483: AIP-38 | Migrate UI to Chakra v3
[Chakra v3 is out ](https://www.chakra-ui.com/blog/00-announcing-v3). We should try to move the new UI to use it to avoid the pain of migrating later. By the time Airflow 3 is released, chakra will certainly have had a few patch releases.

### PR DESCRIPTION
**Pull request by bbovenzi**
Chakra v3 came out last week. Its a really big change. So its better than we migrate sooner rather than later. By the time Airflow 3 comes out it won't be at 3.0.0 anymore.

Closes: https://github.com/apache/airflow/issues/43483

Screenshot 2024-10-31 at 10 13 00 AM
Screenshot 2024-10-31 at 10 13 07 AM
Screenshot 2024-10-31 at 10 12 42 AM
Screenshot 2024-10-31 at 10 12 55 AM

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/ui/src/pages/Dashboard/HealthSection.tsx [hunk 1]
```diff
@@ -16,7 +16,10 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import { Skeleton, Tag, TagLabel, Text, Tooltip } from "@chakra-ui/react";
+import { Skeleton, TagLabel, Text } from "@chakra-ui/react";
+
+import { Tag } from "src/components/ui/Tag";
+import { Tooltip } from "src/components/ui/Tooltip";
```
Comment (ryanahamilton) [inline airflow/ui/src/pages/Dashboard/HealthSection.tsx]: WDYT about adding a `src/components/ui/index.ts` export file to make imports from the snippets cleaner?  It does come with a little extra overhead, but I imagine these won't be constantly changing.
```suggestion
import { Tag, Tooltip } from "src/components/ui/";
```

**Review by pierrejeambrun — COMMENTED** [review]
Great 🎉 

I just noticed a small issue with the switch to `pause/unpause` dags. Request is sent and successful but in the UI it's not correctly displayed, if I refresh things are right. (stays paused or unpaused without refreshing)

**Review by pierrejeambrun — APPROVED** [review]

**Review by jscheffl — APPROVED** [review]
Good to upgrade / switch early!

A lot of UI components are added inthe PR - I assume they are missing (atm) in 3.0.0 and hopefully will be added upstream later such that we can drop the local code?

I also saw a small in-consistency - but maybe already have been before? When I enable/disable DAGs and switch from card to list view, state is not carried over. Also I think some small glitches (not sure if have been like this before) when filtering ans switching card/list view, then filters are reset. Not critical to merge.

LGTM! (Wow a big change!)

**Comment by bbovenzi** [comment]
> Good to upgrade / switch early!
> 
> A lot of UI components are added inthe PR - I assume they are missing (atm) in 3.0.0 and hopefully will be added upstream later such that we can drop the local code?
> 
> I also saw a small in-consistency - but maybe already have been before? When I enable/disable DAGs and switch from card to list view, state is not carried over. Also I think some small glitches (not sure if have been like this before) when filtering ans switching card/list view, then filters are reset. Not critical to merge.
> 
> LGTM! (Wow a big change!)

Chakra v3 switched to open components which gives a lot easier ways to customize them, but also means you need to build a lot yourself. Hence the new `components/ui` directory. Those are all generated from chakra ui snippets. I just commented on an issue on the ChakraUI repo to request their snippets be provided as a separate package so we'd only need to add the components we actually want to customize.

Pierre had a similar issue with the pause/unpause dags. Somehow `rm -rf node_modules && pnpm install` fixed it for him. Let me know if those work for those two issues.

File: airflow/ui/src/components/ui/index.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+export { Alert } from "./Alert";
+export { CloseButton } from "./CloseButton";
+export { InputGroup, type InputGroupProps } from "./InputGroup";
+
+export { Switch } from "./Switch";
+export { Tag } from "./Tag";
+export { Tooltip } from "./Tooltip";
+
+export { ProgressBar } from "./ProgressBar";
+export { Menu } from "./Menu";
```
Comment (ryanahamilton) [inline airflow/ui/src/components/ui/index.ts]: This file might benefit from using wildcard exports, as it is going to continue to grow and input typing exports.
```suggestion
export * from "./Dialog";
export * from "./Pagination";
export * from "./Select";
export * from "./Alert";
export * from "./CloseButton";
export * from "./InputGroup";

export * from "./Switch";
export * from "./Tag";
export *from "./Tooltip";

export * from "./ProgressBar";
export * from "./Menu";
```

File: airflow/ui/src/layouts/Nav/Nav.tsx [hunk 3]
```diff
@@ -43,49 +42,42 @@ export const Nav = () => (
...
         title="Admin"
         to="admin"
       />
     </Flex>
     <Flex flexDir="column">
       <NavButton
-        icon={<FiCornerUpLeft size="1.5rem" />}
-        title="Return to legacy UI"
+        icon={<FiCornerUpLeft size="1.75rem" />}
+        title="Legacy UI"
```
Comment (ryanahamilton) [inline airflow/ui/src/layouts/Nav/Nav.tsx:80]: Nice, I was going to make this suggestion previously.

**Review by ryanahamilton — APPROVED** [review]
Exciting stuff—looks great!

**Comment by jscheffl** [comment]
> Pierre had a similar issue with the pause/unpause dags. Somehow `rm -rf node_modules && pnpm install` fixed it for him. Let me know if those work for those two issues.

Sorry, too late... but comments were not planned to block merge... so all good!

**Pull request merged**